### ISSUE TITLE
Applying "reject" twice on a range of images that contains rejected images will unreject the formerly rejected ones

### ISSUE THREAD

**Issue by codingdave** [darktable-org/darktable#5138]
**Describe the bug**
Given 
* your dt is set up to show all images,
* you have a selection of rejected and unrejected images

Then you press `r`
You will observe that previously unrejected are rejected and rejected are restored to the unrejected state they were in before rejecting.

In other words **what happens is** that rejected and unrejected images are flipped. 
**What the user might expect is** that all images from the selection are rejected.

In contrast take the behavior of the star rating: when you have a selection of images and rate them, all get the same rating. If you press the same rating again, then depending on the setting "rating an image one star twice will not zero out the rating" either they all get 0 stars or nothing happens (This is totally fine).

I propose to add a setting to have a similar feeling for selected images and using the reject key. (I am not sure if this is a bug report or a feature request, because for me its both).

This case has been created on request in #5115.

**Comment by EC1000** [comment]
TL;DR: Could an option "Reject behave like other ratings" be added (FALSE : reject acts as a toggle (current behavior); TRUE : rejecting an already relected image will keep it rejected) to answer different workflow?

---

I totally support this feature request (@codingdave I think you have set it as a bug, it's not, it works as intented, even if it's not the behaviour we would like or expect!). 
In fact I wanted to ask for it since a long time, but when I finally decide to describe the case a few days ago, I first searched for similar subjects, and I discovered the recent changes in #4779 so I took a few days to though about it. 

In addition to the use case describe by @codingdave above (quite annoying indeed), I would like to add another personnal simple use case: I sometimes just reject an already rejected image, because I didn't check the previous rating. 
It may seems stupid, but it happens more than never too me, probably because for other ratings it works. Two examples: 
1. I am hesitating between reject and * or **, going back and forth bewteen a couple of similar images, changing ratings, and then I reject an image: I would like it to be reject (I don't mind/know if it's currently reject or * or **). 
2. I first have rejected just a few images to exclude some bad ones (but did go through all images of my collection). Then later I go through this collection to rate all images (usually rejected/** on a first pass for me). 

I will just look at the image and use keyboard shortcut, not look at the rate, and with the current behavior (or the old one too), I will end up "un-rejecting" an image I wanted to "re-reject". 

I understand there are different workflows. I think it can probably be summarize to the way you see `reject`:
1. As a rating, -1 (obviously with a particular meaning, a dedicated icon, filter, but still a rating), it is in particular the view defined by IPTC standards ("[xmp:Rating=-1 indicates reject](https://www.iptc.org/std/photometadata/specification/IPTC-PhotoMetadata#image-rating)")
2. As dedicated status, hence the toggle behavior (some softwares (AfterShot, Lightroom) even seem to use a independant status not linked to rating with reject/neutral/pick values (or -1/0/1 behind), I must say I totally dislike that because it's two times more actions, and also not standard; dt behavior is a little bit more in between).

I don't know if it would be complicated or not: could it be possible to add an option "Reject behave like other ratings"?
- FALSE : reject acts as a toggle (current behavior), 
- TRUE : rejecting an already rejected image will keep it rejected

**Comment by EC1000** [comment]
> I propose to add a setting to have a similar feeling for selected images and using the reject key. (I am not sure if this is a bug report or a feature request, because for me its both).

Changing the title to something like "FR: add a option for reject behavior" would probably be clearer.

**Comment by codingdave** [comment]
I did not realize before, that the reject button was

1. always toggling
2. did not restore the old rating on the second reject that unrejected the image and that restoring the old rating has been implemented in the last months since the 3.0 release

I was investigating about that because I thought the behavior got worse since the last release but I can confirm I see the improvement. 
However, the reject behavior does not make it easy for me to go through my images and applying rejects because I have to check every time if all selected images are already rejected or not and have to react on that. (Of course I can filter to see only non-rejected ones but there are several reasons I am not doing that, like I might want to see what similar images I already have filtered out or because of buggy jumps on the culling/file manager views).

Thinking about this for a long time I personally came to the conclusion the best option was a checkbox that offers the user the choice over the toggle or the "reject always wins" behavior. Is there any argument against that? Or any better solution I did not yet came upon?

**Comment by parafin** [comment]
I would say reject should only reject images (not toggle rejected status). If I remember correctly we have now "unreject" feature that restores original rating, I think this is a separate action which needs its own key binding and button. As a compromise to save space in UI reject button can automatically change to unreject when only rejected images are selected.

Rationale - I don't see a scenario where user wants to toggle rejected state on a mixed selection of rejected and not rejected images.

**Comment by ptilopteri** [comment]
there is already a "check box" for reject/unreject available on the thumb-nails in light table view.

**Comment by codingdave** [comment]
> there is already a "check box" for reject/unreject available on the thumb-nails in light table view.

@ptilopteri I think you are talking about the button showing a black/red "x", a checkbox is e.g. in the options the element for storing "rating an image one star twice will not zero out the rating" that you can check/uncheck.

@parafin I agree that reject and unreject are 2 different operations. The thumbnail already has 2 buttons (black/red x), but both are triggered by the keyboard alternatively, making the key<r> a toggle button for each image in the (1 to n-elementary) selection. Since the logic is applied to each element in the selection there is an implicit, hard to predict logic going on.

Having 2 keyboard shortcuts, 1 for rejecting, 1 for un-rejecting/restoring old rating would solve that issue straightforward because it removes the implicit logic and gives the user back full explicit control of what is happening. So I propose to have something like r for reject and Shift-r for un-rejecting, in analogy to group (g) and ungroup (Shift-g).
